micro/service.go

```diff
@@ -90,9 +91,10 @@ type (
 
 	// ServiceIdentity contains fields helping to identity a service instance.
 	ServiceIdentity struct {
-		Name    string `json:"name"`
-		ID      string `json:"id"`
-		Version string `json:"version"`
+		Name     string            `json:"name"`
+		ID       string            `json:"id"`
+		Version  string            `json:"version"`
+		Metadata map[string]string `json:"metadata"`
```

Comment:
Maybe we could use `omitempty` for metadata to avoid sending `null` field if it's not provided?

Reply:
updated it in both places.

Reply:
(I actually believe the same)